tests/run_tests.sh

```diff
@@ -24,17 +24,24 @@ for PYTHON in /opt/python/*/bin/python; do
 	$PYTHON $MY_DIR/manylinux-check.py ${AUDITWHEEL_POLICY} ${AUDITWHEEL_ARCH}
 	# Make sure that SSL cert checking works
 	$PYTHON $MY_DIR/ssl-check.py
-	# Make sure sqlite3 module can be loaded properly and is the manylinux version one
-	# c.f. https://github.com/pypa/manylinux/issues/1030
-	$PYTHON -c 'import sqlite3; print(sqlite3.sqlite_version); assert sqlite3.sqlite_version_info[0:2] >= (3, 34)'
-	# pythonx.y & pipx.y shall be available directly in PATH
+	IMPLEMENTATION=$(${PYTHON} -c "import sys; print(sys.implementation.name)")
 	PYVERS=$(${PYTHON} -c "import sys; print('.'.join(map(str, sys.version_info[:2])))")
-	LINK_VERSION=$(python${PYVERS} -V)
+	if [ "${IMPLEMENTATION}" == "pypy" ]; then
+		LINK_PREFIX=pypy
+	else
+		LINK_PREFIX=python
+		# Make sure sqlite3 module can be loaded properly and is the manylinux version one
+		# c.f. https://github.com/pypa/manylinux/issues/1030
+		$PYTHON -c 'import sqlite3; print(sqlite3.sqlite_version); assert sqlite3.sqlite_version_info[0:2] >= (3, 34)'
+		# pipx.y shall be available directly in PATH
+		LINK_VERSION=$(pip${PYVERS} -V)
+		REAL_VERSION=$(${PYTHON} -m pip -V)
+		test "${LINK_VERSION%% from *}" = "${REAL_VERSION%% from *}"
+	fi
```

Comment:
Shouldn't these sanity checks be run on PyPy too? If they fail there is a problem with the PyPy build ...

Reply:
can't run the sanity check of the `pipx.y` symlink in `/usr/local/bin` since there are no symlink for PyPy pip in there. As mentioned in an earlier comment, going to remove those anyway.

For sqlite3, this is specific to cpython being rebuilt from source with a specific version (well should be specific, this has been an issue before). Given PyPY comes as a full featured binary, this check isn't right (& fails because the embedded sqlite3 is version 3.28 or 3.29 - I don't remember exactly)

Reply:
> the embedded sqlite3 is version 3.28 or 3.29

Hmm. Is it documented somewhere what sqlite3 a specific python version expects? If cpython is statically linking to 3.35 that would suggest PyPy should as well.

Reply:
The minimum sqlite3 version required by cpython is fairly low: https://github.com/python/cpython/blob/220dd80a2671f57486055955d5422163cf73ed33/Modules/_sqlite/module.c#L33

The default version in manylinux1 was even older than this requirement though:
- https://github.com/pypa/manylinux/issues/90
- https://github.com/pypa/manylinux/pull/93

It's been updated since then in order not to get issue reported because of CVE on older versions.

Windows/macOS installers of CPython also update sqlite3 regularly as noted in their release notes.